processor/probabilisticsamplerprocessor/sampler_mode.go

```diff
@@ -0,0 +1,316 @@
+	rprime14 := numHashBuckets - 1 - hashed
+
+	// There are 18 unused bits from the FNV hash function.
+	unused18 := uint64(hashed32 >> (32 - numHashBucketsLg2))
+	mixed28 := unused18 ^ (unused18 << 10)
+
+	// The 56 bit quantity here consists of, most- to least-significant:
+	// - 14 bits: R' = numHashBuckets - 1 - hashed
+	// - 28 bits: mixture of unused 18 bits
+	// - 14 bits: original `hashed`.
+	rnd56 := (rprime14 << 42) | (mixed28 << 14) | hashed
+
+	// Note: by construction:
+	// - OTel samplers make the same probabilistic decision with this r-value,
+	// - only 14 out of 56 bits are used in the sampling decision,
+	// - there are only 32 actual random bits.
+	rnd, _ := sampling.UnsignedToRandomness(rnd56)
+	return rnd
+}
+
+// consistencyCheck checks for certain inconsistent inputs.
+//
+// if the randomness is missing, returns ErrMissingRandomness.
+func consistencyCheck(rnd randomnessNamer, _ samplingCarrier) error {
+	if isMissing(rnd) {
+		return ErrMissingRandomness
+	}
+	return nil
+}
+
+// makeSample constructs a sampler. There are no errors, as the only
+// potential error, out-of-range probability, is corrected automatically
+// according to the README, which allows percents >100 to equal 100%.
+//
+// Extending this logic, we round very small probabilities up to the
+// minimum supported value(s) which varies according to sampler mode.
+func makeSampler(cfg *Config) dataSampler {
+	// README allows percents >100 to equal 100%.
+	pct := cfg.SamplingPercentage
+	if pct > 100 {
+		pct = 100
+	}
+
+	never := &neverSampler{}
+
+	if pct == 0 {
+		return never
+	}
+
+	// Note: the original hash function used in this code
+	// is preserved to ensure consistency across updates.
+	//
+	//   uint32(pct * percentageScaleFactor)
+	//
+	// (a) carried out the multiplication in 32-bit precision
+	// (b) rounded to zero instead of nearest.
+	scaledSamplerate := uint32(pct * percentageScaleFactor)
+
+	if scaledSamplerate == 0 {
+		return never
+	}
+
+	// Convert the accept threshold to a reject threshold,
+	// then shift it into 56-bit value.
+	reject := numHashBuckets - scaledSamplerate
+	reject56 := uint64(reject) << 42
+
+	threshold, _ := sampling.UnsignedToThreshold(reject56)
+
+	return &hashingSampler{
+		tvalueThreshold: threshold,
+		hashSeed:        cfg.HashSeed,
+
+		// Logs specific:
+		logsTraceIDEnabled:            cfg.AttributeSource == traceIDAttributeSource,
+		logsRandomnessSourceAttribute: cfg.FromAttribute,
+	}
+}
+
+// randFunc returns randomness (w/ named policy), a carrier, and the error.
+type randFunc[T any] func(T) (randomnessNamer, samplingCarrier, error)
+
+// priorityFunc makes changes resulting from sampling priority.
+type priorityFunc[T any] func(T, randomnessNamer, sampling.Threshold) (randomnessNamer, sampling.Threshold)
+
+// commonSamplingLogic implements sampling on a per-item basis
+// independent of the signal type, as embodied in the functional
+// parameters:
+func commonSamplingLogic[T any](
```

Comment:
Good call! I've named it `commonShouldSampleLogic()` and inverted the return, so that now the callers are in a `RemoveIf(... !commonShouldSampleLogic())`.